From 4c138687008a3d212906367a315ea79b6f6727c9 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Thu, 29 Jun 2017 20:54:08 +0200
Subject: [PATCH] tests/tools: fake $PWD so that is starts with ///.

This is needed to skip overzealous /tmp checks in afl-cmap.

Fixes:

    ======================================================================
    ERROR: tests.test_cmin.test
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
        self.test(*self.arg)
      File "/tmp/.../tests/test_cmin.py", line 68, in test
        b'1',
      File "/tmp/.../tests/test_cmin.py", line 51, in run_afl_cmin
        run(cmdline)
      File "/tmp/.../tests/tools.py", line 148, in run
        raise ipc.CalledProcessError(child.returncode, cmd[0])
    CalledProcessError: Command 'py-afl-cmin' returned non-zero exit status 1
    -------------------- >> begin captured stdout << ---------------------
    ...
    [-] Error: do not use this script in /tmp or /var/tmp.
---
 tests/tools.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/tools.py b/tests/tools.py
index 0c9240c..f3ebab1 100644
--- a/tests/tools.py
+++ b/tests/tools.py
@@ -131,6 +131,7 @@ def clean_environ():
     os.environ['AFL_SKIP_CPUFREQ'] = '1'
     os.environ['AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES'] = '1'
     os.environ['AFL_NO_AFFINITY'] = '1'
+    os.environ['PWD'] = '//' + os.getcwd()
 
 def run(cmd, stdin='', xstatus=0):
     child = ipc.Popen(
